### PR TITLE
feat(input): added drop event support

### PR DESCRIPTION
**Pull request by XFree**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
drop event not support

**What is the new behavior (if this is a feature change)?**
added drop event support 

**Does this PR introduce a breaking change?**
no

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [x] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

File: src/ng/directive/input.js [hunk 1]
```diff
@@ -1308,7 +1308,7 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
 
     // if user modifies input value using context menu in IE, we need "paste" and "cut" events to catch it
```
Comment (gkalpak) [inline src/ng/directive/input.js]: This comment needs updating now :smiley:

File: src/ng/directive/input.js [hunk 2]
```diff
@@ -1308,7 +1308,7 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
...
     if ($sniffer.hasEvent('paste')) {
```
Comment (gkalpak) [inline src/ng/directive/input.js]: Do we have to check for `drop` event support separately? Are there browsers that support one but not the other?

Comment (petebacondarwin) [reply src/ng/directive/input.js]: I can't imagine that they would support drop but not paste. But perhaps the other way around?

**Comment by XFree** [comment]
The assembly has fallen with the comment "I have a colon after the closing parenthesis :): but there are colon ones?

File: src/ng/directive/input.js (hunk 1 above)
Comment (XFree) [reply src/ng/directive/input.js]: Thank's.

**Comment by petebacondarwin** [comment]
@Xfree - hmm, it seems that the commit message check is picking up a commit that is from before your PR: `style(css) separate selectors and declarations by new lines`.

Perhaps try rebasing on master, if you haven't recently?

**Comment by XFree** [comment]
@petebacondarwin, done

File: src/ng/directive/input.js (hunk 2 above)
Comment (petebacondarwin) [reply src/ng/directive/input.js]: From caniuse.com, it looks like support for event drop is equivalent to support for the paste event.

**Pull request merged**